### PR TITLE
daniel_dominguez Initial commit

### PR DESCRIPTION
**Pull request by dcd000**
Here you have files associated to session-01, exercises 01, 02 and 03.

### CODE REVIEW

File: daniel_dominguez/session01/exercise01/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: wordpress
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
```
[the remark this answers is not in the document]
Comment (dcd000) [reply daniel_dominguez/session01/exercise01/ingress.yaml]: Hello Juan.
Yes it does, as if you are confortable with the default values you don't need to specify them.

Here you have the the upstream definition in nginx.conf:    

upstream sticky-exercise-01-wordpress-80 {
        sticky hash=md5 name=INGRESSCOOKIE  httponly;

        keepalive 32;

        server 10.32.0.13:80 max_fails=0 fail_timeout=0;

    }

The point is that if the upstream only has one endpoint, nginx does not sent set-Cookie response header.

Example 1: HTTP Response headers with **1 pod** (without set-Cookie):
HTTP/1.1 200 OK
Server: nginx/1.13.8
Date: Thu, 10 May 2018 15:11:16 GMT
Content-Type: text/html; charset=UTF-8
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
X-Powered-By: PHP/7.0.30
Link: <http://wordpress-01.35.171.157.107.nip.io/wp-json/>; rel="https://api.w.org/"
Vary: Accept-Encoding

Example 2: HTTP Response headers with **2 pods** (with set-Cookie):
HTTP/1.1 200 OK
Server: nginx/1.13.8
Date: Thu, 10 May 2018 15:13:20 GMT
Content-Type: text/html; charset=UTF-8
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
**Set-Cookie: INGRESSCOOKIE=d7f1a72d02b8c4279203e3a08bfd73fe; Path=/; HttpOnly**
X-Powered-By: PHP/7.0.30
Link: <http://wordpress-01.35.171.157.107.nip.io/wp-json/>; rel="https://api.w.org/"
Vary: Accept-Encoding

File: daniel_dominguez/session01/exercise02/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: wordpress
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
```
[the remark this answers is not in the document]
Comment (dcd000) [reply daniel_dominguez/session01/exercise02/ingress.yaml]: See above

File: daniel_dominguez/session01/exercise03/ingress-tls.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    - wordpress-03.35.171.157.107.nip.io
+    secretName: wordpress-tls
+  rules:
+  - host: wordpress-03.35.171.157.107.nip.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: wordpress
+          servicePort: 80
```
[the remark this answers is not in the document]
Comment (dcd000) [reply daniel_dominguez/session01/exercise03/ingress-tls.yaml]: Done in ce64de0b704fb93fb6674366782ce1e9c91e02f7

File: daniel_dominguez/session01/exercise01/ingress.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (juan131) [reply daniel_dominguez/session01/exercise01/ingress.yaml]: Great explanation! Thanks

File: daniel_dominguez/session01/exercise02/ingress.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (juan131) [reply daniel_dominguez/session01/exercise02/ingress.yaml]: 👍

**Pull request merged**